### PR TITLE
Log errors, return user->_id from DB on oauth

### PR DESCRIPTION
**Pull request by talyguryn**

### CODE REVIEW

**Review by n0str — APPROVED** [review]

File: www/components/base/models/BaseException.php [hunk 1]
```diff
@@ -5,15 +5,11 @@
...
-    protected $logger;
-
+class BaseException extends Exception
+{
     public function __construct($message, $code = 0, $previous = null)
     {
-        $this->logger = Log::instance();
-
-        $this->logger->warning($message);
+        Log::instance()->debug($message);
```
Comment (neSpecc) [inline www/components/base/models/BaseException.php]: This is not a debug level, probably you should use error-level or warning-level

File: www/components/oauth/OAuth.php [hunk 2]
```diff
@@ -71,7 +71,8 @@ public function code(Request $req, Response $res, $args)
             'google_id' => $userData['google_id'],
             'photo' => $userData['photo'],
             'name' => $userData['name'],
-        ], self::generateSignatureKey($userData['google_id']));
+            'user_id' => $user->id,
```
Comment (neSpecc) [inline www/components/oauth/OAuth.php]: Here is the other task

File: www/components/oauth/OAuth.php [hunk 3]
```diff
@@ -49,29 +49,35 @@ public function code(Request $req, Response $res, $args)
         $profileInfo = HTTP::request('GET', self::GOOGLE_PROFILE_URL, [], [$header]);
         $profileInfo = @json_decode($profileInfo);
 
-        if (!is_null($profileInfo->error)) {
-            Log::instance()->warning('Google OAuth failed. Reason: ' . $profileInfo->error->reason);
-            return $res->withStatus(HTTP::CODE_SERVER_ERROR, $profileInfo->error->reason);
+        Log::instance()->debug('Profile info from Google: ' . json_encode($profileInfo));
+
+        if (isset($profileInfo->error)) {
```
Comment (neSpecc) [inline www/components/oauth/OAuth.php]: `!empty` would be better

File: www/components/oauth/OAuth.php [hunk 4]
```diff
@@ -49,29 +49,35 @@ public function code(Request $req, Response $res, $args)
...
             'photo' => $profileInfo->picture,
-            'name' => $profileInfo->name
         ];
 
         /** Save new user to database */
-        $user = new User($userData['id']);
+        $user = new User();
         $user->sync($userData);
 
+        Log::instance()->debug('User model from DB: ' . json_encode($user));
```
Comment (neSpecc) [inline www/components/oauth/OAuth.php]: `User model from DB ` -> `New user registered`

Comment (neSpecc) [reply www/components/oauth/OAuth.php]: or user updating also handles here?

File: www/components/oauth/OAuth.php [hunk 5]
```diff
@@ -49,29 +49,35 @@ public function code(Request $req, Response $res, $args)
...
+
         $jwt = JWT::encode([
             'iss' => Config::get('JWT_ISS'),
             'aud' => Config::get('JWT_AUD'),
             'iat' => time(),
-            'google_id' => $userData['google_id'],
+//            'google_id' => $userData['google_id'],
```
Comment (neSpecc) [inline www/components/oauth/OAuth.php]: are u sure that we don't need this on the client-side?

File: www/public/index.php [hunk 6]
```diff
@@ -34,20 +35,47 @@
...
+            ->withStatus(404)
+            ->withHeader('Content-Type', 'text/html')
+            ->write('Page not found');
+    };
+};
+
+/**
+ * Enable Hawk Catcher
+ */
+\Hawk\HawkCatcher::instance(Config::get('HAWK_TOKEN'));
```
Comment (neSpecc) [inline www/public/index.php]: wrap with `!empty(Config::get('HAWK_TOKEN'))`

File: www/schema/types/Mutation.php [hunk 7]
```diff
@@ -34,32 +34,32 @@ public function __construct()
...
-
-                            $selectedFields = $info->getFieldSelection();
-                            if (in_array('folders', $selectedFields)) {
-                                $user->fillFolders();
-                            }
-
-                            return $user;
-                        }
-                    ],
+//                    'user' => [
```
Comment (neSpecc) [inline www/schema/types/Mutation.php]: User mutation will be used to update photos, name and other Settings, that User will made with the Client

**Review by neSpecc — CHANGES_REQUESTED** [review]

**Review by actorFromCalifornia — APPROVED** [review]

File: www/components/middleware/Auth.php [hunk 8]
```diff
@@ -24,29 +30,49 @@ class Auth
...
-        try {
-            $decoded = JWT::decode($token, $key, ['HS256']);
-            $GLOBALS['user'] = (array) $decoded;
-        } catch (\Exception $e) {
-            Log::instance()->notice("Auth for {$payload->google_id} failed because of {$e->getMessage()}");
+            $payload = JWT::jsonDecode(JWT::urlsafeB64Decode($jwtParts[1]));
+
+            $key = OAuth::generateSignatureKey($payload->user_id);
+
+            $decoded         = JWT::decode($token, $key, ['HS256']);
```
Comment (neSpecc) [inline www/components/middleware/Auth.php]: лишний отступ

**Review by neSpecc — APPROVED** [review]

**Review by neSpecc — CHANGES_REQUESTED** [review]
Wait for testing

**Review by neSpecc — CHANGES_REQUESTED** [review]
[2018-01-17 13:24:24.419692] [warning] JWT is missing
[2018-01-17 13:24:25.494563] [warning] JWT is missing
[2018-01-17 13:24:27.427426] [warning] JWT is missing
[2018-01-17 13:24:35.205452] [warning] JWT is missing
[2018-01-17 13:24:42.080646] [warning] Undefined property: stdClass::$user_id
[2018-01-17 13:24:42.095460] [error] Undefined property: stdClass::$user_id in /var/www/codex.notes.server/public/index.php:59
#0 /var/www/codex.notes.server/Components/Middleware/Auth.php(62): App\{closure}(8, 'Undefined prope...', '/var/www/codex....', 62, Array)
#1 [internal function]: App\Components\Middleware\Auth->jwt(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\Route))
#2 /var/www/codex.notes.server/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Array, Array)
#3 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\Route))
#4 /var/www/codex.notes.server/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\Route))
#5 /var/www/codex.notes.server/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\Route->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#6 /var/www/codex.notes.server/vendor/slim/slim/Slim/Route.php(313): Slim\Route->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))
#7 /var/www/codex.notes.server/vendor/slim/slim/Slim/App.php(495): Slim\Route->run(Object(Slim\Http\Request), Object(Slim\Http\Response))
#8 /var/www/codex.notes.server/vendor/akrabat/rka-ip-address-middleware/src/IpAddress.php(93): Slim\App->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))
#9 [internal function]: RKA\Middleware\IpAddress->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#10 /var/www/codex.notes.server/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(RKA\Middleware\IpAddress), Array)
#11 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#12 /var/www/codex.notes.server/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#13 /var/www/codex.notes.server/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\App->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#14 /var/www/codex.notes.server/vendor/slim/slim/Slim/App.php(388): Slim\App->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))
#15 /var/www/codex.notes.server/vendor/slim/slim/Slim/App.php(296): Slim\App->process(Object(Slim\Http\Request), Object(Slim\Http\Response))
#16 /var/www/codex.notes.server/public/index.php(94): Slim\App->run()
#17 {main}

**Review by neSpecc — APPROVED** [review]

**Pull request merged**